api/valor_api/backend/metrics/metric_utils.py

```diff
@@ -253,6 +256,109 @@ def create_metric_mappings(
     return ret
 
 
+def log_evaluation_analytics(
+    db: Session,
+    evaluation_id: int,
+    prediction_filter: schemas.Filter,
+    groundtruth_filter: schemas.Filter,
+):
+    """
+    Store analytics regarding the evaluations execution in the metadata field of the evaluation table.
+
+    Parameters
+    ----------
+    db : Session
+        The database Session to query against.
+    evaluation_id : int
+        The job ID to create metrics for.
+    prediction_filter : schemas.Filter
+        The filter to be used to query predictions.
+    groundtruth_filter : schemas.Filter
+        The filter to be used to query groundtruths.
+    """
+    # get ground truth, prediction, annotation, and label counts
+    gt_subquery = (
+        Query(
+            models.GroundTruth,
+        )
+        .filter(groundtruth_filter)
+        .groundtruths(as_subquery=False)
+        .alias()
+    )
+
+    gts = db.execute(
+        select(
+            gt_subquery.c.annotation_id,
+            gt_subquery.c.label_id,
+        ).select_from(gt_subquery)
+    ).all()
+
+    # handle edge case where no gts come back
+    if not gts:
+        gt_annotation_id, gt_label_id = set(), set()
+    else:
+        gt_annotation_id, gt_label_id = map(set, zip(*gts))
```

Comment:
Charles: does this process make sense to you? I was trying to minimize the number of psql calls (relative to querying the number of datums separate from the number of labels, etc.)

Reply:
Im not sure if this is possible, but the following would reduce the queries to 2.

```
db.query(
    Query(
        func.count(distinct(models.Datum.id)),
        func.count(distinct(models.Annotation.id)),
        func.count(distinct(models.Label.id)),
    )
    .filter(groundtruth_filter)
    .groundtruth()
).all()
```